machinery/topology.go

```diff
@@ -111,12 +120,16 @@ func NewTopology(options ...TopologyOptionsFunc) *Topology {
 
 	addPoliciesToGraph(graph, policies)
 
+	if !o.AllowLoops && !isDAG(graph) {
+		return nil, errors.New("loop detected in graph check linking functions")
```

Comment:
I know we've been back and forth about returning `nil` or the generated topology containing loops, along with the error. If we want to let users decide how to handle the error in this case (e.g. save a copy of the invalid topology somewhere, then raise the error), I guess we need to get back to that original design where the new topology is returned regardless, no?